EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -78,40 +80,77 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch
             TrainTestData trainTestSplit = mlContext.Data.TrainTestSplit(dataView, testFraction: 0.1, null, 42);
             IDataView trainingData = trainTestSplit.TrainSet;
             IDataView testData = trainTestSplit.TestSet;
-
             var trainer = mlContext.BinaryClassification.Trainers.FastTree(labelColumnName: "Label", featureColumnName: "Features");
-
             var pipeline = mlContext.Transforms.Concatenate("Features", trainingVariables)
                 .Append(mlContext.BinaryClassification.Trainers.FastTree(labelColumnName: "Label", featureColumnName: "Features"));
+            var trainedModel = pipeline.Fit(trainingData); //this is NOT thread safe
+            var predictionEngine = mlContext.Model.CreatePredictionEngine<PsmData, TruePositivePrediction>(trainedModel);//this is NOT thread safe
 
-            var trainedModel = pipeline.Fit(trainingData);
+            int maxThreads = fileSpecificParameters.FirstOrDefault().fileSpecificParameters.MaxThreadsToUsePerFile;
+            int[] threads = Enumerable.Range(0, maxThreads).ToArray();
+            object lockObject = new object();
 
-            var predictionEngine = mlContext.Model.CreatePredictionEngine<PsmData, TruePositivePrediction>(trainedModel);
+            if (outputFolder != null)
```

Comment:
I understand what you're doing here, but it's pretty hacky to use an output folder to determine parallelization. Please comment what's going on here and at the class variable OutputFolder.
It makes more sense to make these methods non-static and just reference the OutputFolder object rather than passing these in.

Reply:
i think it's a good idea to have the task have an OutputFolder variable, in case that's needed for future code. it is a property of the task

Reply:
comment explaining rationale added to code